### ISSUE TITLE
Upload On S3 with Symfony 3.4

### ISSUE THREAD

**Issue by AlanLima91** [dustin10/VichUploaderBundle#1038]
Hi, I'm trying since several month to upload a file on a S3 (AWS), with a Symfony 3.4 API that I've dvelopp.

I've tried to fix it by myself with many tutorials/resources/docs and I'm now facing a wall.. If someone can tell what I'm missing out, I'll be grateful. 

Here's my conf:

composer.json
```JSON
"require": {
        "php": ">=5.5.9",
        "aws/aws-sdk-php": "^3.87",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/orm": "^2.5",
        "friendsofsymfony/rest-bundle": "^2.4",
        "incenteev/composer-parameter-handler": "^2.0",
        "knplabs/knp-gaufrette-bundle": "^0.5.3",
        "league/flysystem-aws-s3-v3": "^1.0",
        "nelmio/api-doc-bundle": "^3.3",
        "nelmio/cors-bundle": "^1.5",
        "oneup/flysystem-bundle": "^3.0",
        "sensio/distribution-bundle": "^5.0.19",
        "sensio/framework-extra-bundle": "^5.0.0",
        "symfony/form": "3.4.*",
        "symfony/monolog-bundle": "^3.1.0",
        "symfony/polyfill-apcu": "^1.0",
        "symfony/serializer": "3.4.*",
        "symfony/swiftmailer-bundle": "^2.6.4",
        "symfony/symfony": "3.4.*",
        "twig/twig": "^1.0||^2.0",
        "vich/uploader-bundle": "^1.8",
        "white-october/pagerfanta-bundle": "^1.2"
    }
```

vich / gaufrette
```yaml
knp_gaufrette:
    stream_wrapper: ~
    adapters:
        document_adapter:
            aws_s3:
                service_id: ct_file_store.s3
                bucket_name: '%AWS_BUCKET_NAME%'
                detect_content_type: true
                options:
                    create: true
                    directory: uploads/landingpage
    filesystems:
        document_fs:
            adapter:    document_adapter

vich_uploader:
    db_driver: orm
    storage: gaufrette
    mappings:
        document:
            inject_on_load: true
            uri_prefix: "%AWS_BASE_URL%%AWS_BUCKET_NAME%/document"
            upload_destination: document_fs
            namer: vich_uploader.namer_uniqid
            delete_on_update:   true #should the file be deleted when a new file is uploaded
            delete_on_remove:   true #should the file be deleted when the entity is removed
```
Service.yml
```yaml
ct_file_store.s3:
        class: Aws\S3\S3Client
        factory: [Aws\S3\S3Client, 'factory']
        arguments:
            -
                version: '2006-03-01'
                region: '%AWS_S3_REGION%'
                credentials: { key: '%AWS_KEY%', secret: '%AWS_SECRET_KEY%' }
```
My Entity
```php
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;

/**
 * LandingPage
 *
 * @ORM\Table(name="landing_page")
 * @ORM\Entity(repositoryClass="AppBundle\Repository\LandingPageRepository")
 * @Vich\Uploadable
 */
class LandingPage
{
    /**
     * @ORM\Column(name="image", type="string", length=190, nullable=true)
     *
     * @var string
     */
    private $image;

    /**
     * @Vich\UploadableField(mapping="document", fileNameProperty="image")
     *
     * @var File
     */
    private $imageFile;

    /**
     * @ORM\Column(name="updated_at", type="datetime", nullable=true)
     */
    private $updatedAt;

    /**
     * @param string $image
     *
     * @return LandingPage
     */
    public function setImage($image)
    {
        $this->image = $image;

        return $this;
    }

    /**
     * @return string|null
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * If manually uploading a file (i.e. not using Symfony Form) ensure an instance
     * of 'UploadedFile' is injected into this setter to trigger the  update. If this
     * bundle's configuration parameter 'inject_on_load' is set to 'true' this setter
     * must be able to accept an instance of 'File' as the bundle will inject one here
     * during Doctrine hydration.
     *
     * @param File|\Symfony\Component\HttpFoundation\File\UploadedFile $image
     *
     * @return LandingPage
     */
    public function setImageFile(UploadedFile $image = null)
    {
        $this->imageFile = $image;

        if ($image) {
            $this->updatedAt = new \DateTime();
        }

        return $this;
    }

    /**
     * Returns the image to be uploaded.
     *
     * @return File|null
     */
    public function getImageFile()
    {
        return $this->imageFile;
    }
}

```
My Form
```php
<?php

namespace AppBundle\Form;

use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\FormBuilderInterface;
use Symfony\Component\OptionsResolver\OptionsResolver;
use Vich\UploaderBundle\Form\Type\VichImageType;

class LandingPageType extends AbstractType
{
    /**
     * {@inheritdoc}
     */
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder->add('imageFile', VichImageType::class, [
            'required' => true,
            'allow_delete' => true,
            'by_reference' => false,
            'allow_file_upload' => true,
        ]);
    }
}
```

The response from the my API : 
```JSON
{
    "code": 400,
    "message": "Validation Failed",
    "errors": {
        "children": {
            "imageFile": {
                "errors": [
                    "This value is not valid."
                ],
                "children": {
                    "file": []
                }
            }
        }
    }
}
```

**Comment by bastijan** [comment]
I have the same issue... Did you solved?

**Comment by AlanLima91** [comment]
Yes I've solved it by updating to Symfony 4.4

**Comment by garak** [comment]
Closing, since Symfony 3.4 is EOLed
